lib/wca_live/competitions.ex

```diff
@@ -58,12 +58,32 @@ defmodule WcaLive.Competitions do
   @spec get_competition_by_wca_id!(String.t()) :: %Competition{}
   def get_competition_by_wca_id!(wca_id), do: Repo.get_by!(Competition, wca_id: wca_id)
 
+  @spec fetch_competition(any()) :: {:error, any()} | {:ok, %{optional(atom()) => any()}}
   @doc """
-  Gets a single competition.
+  Gets a single competition
   """
   @spec fetch_competition(term()) :: {:ok, %Competition{}} | {:error, Ecto.Queryable.t()}
   def fetch_competition(id), do: Repo.fetch(Competition, id)
 
+  @doc """
+  Gets a single competition by numeric id or wca id
+  """
+  @spec fetch_competition_by_id_or_wca_id(term()) ::
+          {:ok, %Competition{}} | {:error, Ecto.Queryable.t()}
+  def fetch_competition_by_id_or_wca_id(id) do
+    case parse_competition_id_or_wca_id(id) do
+      {:competition_id, competition_id} -> Repo.fetch(Competition, competition_id)
+      {:wca_id, wca_id} -> Repo.fetch(Ecto.Query.where(Competition, wca_id: ^wca_id))
+    end
+  end
```

Comment:
I would do the parsing and conditional call in controller. We already have `fetch_competition` above, so we just need this:

```suggestion
  @doc """
  Gets a single competition by WCA id.
  """
  @spec fetch_competition_by_wca_id(String.t()) ::
          {:ok, %Competition{}} | {:error, Ecto.Queryable.t()}
  def fetch_competition_by_wca_id(id) do
    Repo.fetch(where(Competition, wca_id: ^wca_id))
  end
```

The main reason is that I don't like passing an "unstructured" id string and parsing it here.